TailLight/device.cpp

```diff
@@ -1,52 +1,131 @@
+        GUID_DEVINTERFACE_HID));
+
+    {     
+        // Assumption: Device will arrive before removal.
+        if (IsEqualGUID(pDevInterface->Event,
+            GUID_DEVICE_INTERFACE_ARRIVAL)) {
+
+            // Opening a device may trigger PnP operations. Ensure that either a
+            // timer or a work item is used when opening up a device.
+            // Refer to page 356 of "Programming The Microsoft Windows Driver
+            // Model", 2nd edition by Walter Oney and IoGetDeviceObjectPointer.
+            // 
+            // Also note that eventually several system threads will be running
+            // in parallel. As a PnP system thread must not block, serializing
+            // would require a queue.
+            //
+            // NOTE: It is possible for us to get blocked waiting for a system
+            // thread. One solution would be to use a timer that spawns a
+            // system thread on timeout vs. a wait loop.
+            //
+            // What the operating system does with the return value is
+            // undocumented. Since the docs say to return success or an error
+            // value, follow the spec.
+            return CreateWorkItemForIoTargetOpenDevice((WDFDEVICE)pvContext,
+                *pDevInterface->SymbolicLinkName);
+        }
     }
 
+    return STATUS_SUCCESS;
+}
+
+
+NTSTATUS EvtSelfManagedIoInit(WDFDEVICE device) 
+/*++
+Routine Description:  
+
+    Starts the enumeration of HID class device interfaces. Since drivers enable
+    their device interfaces after they initialize. Thus, if the device is able,
+    it will be able to process our requests.
+
+Arguments:
+
+    device - Our (TailLight's) WDF device.
+
+Returns:
+
+    Result of attempting to start device interface enumeration.
+--*/
+{
+
+    WDFDRIVER driver = WdfDeviceGetDriver(device);
+    DEVICE_CONTEXT* pDeviceContext = WdfObjectGet_DEVICE_CONTEXT(device);
+
+    NTSTATUS status = STATUS_SUCCESS;
+
+    status = IoRegisterPlugPlayNotification(
+        EventCategoryDeviceInterfaceChange,
+        PNPNOTIFY_DEVICE_INTERFACE_INCLUDE_EXISTING_INTERFACES,
+        (PVOID)&GUID_DEVINTERFACE_HID,
+        WdfDriverWdmGetDriverObject(driver),
+        PnpNotifyDeviceInterfaceChange,
+        (PVOID)device,
+        &pDeviceContext->pnpDevInterfaceChangedHandle
+    );
+
     return status;
 }
 
 
-UNICODE_STRING GetTargetPropertyString(WDFIOTARGET target, DEVICE_REGISTRY_PROPERTY DeviceProperty) {
+UNICODE_STRING GetTargetPropertyString(WDFIOTARGET target, DEVICE_REGISTRY_PROPERTY DeviceProperty) 
+/*++
+Routine Description:
+
+    Returns a device property string about the IO target. These are the same
+    strings that are shown in the properties pages' details tab for a deice.
+
+Arguments:
+
+    target - An IO target to be interrogated.
+
+    DeviceProperty - The device property to be retrieved.
+
+Returns:
+
+    Information about or describing the device bound
+    to the IO target.
+--*/
```

Comment:
You're not changing the `GetTargetPropertyString` function in this PR. Therefore, please don't introduce doc updates to the function in this PR.

You're free to propose doc. updates to the function, but that should be in a _separate_ PR.